File: Backend/src/main/java/com/bookstore/dto/GoogleSignInRequest.java
package com.bookstore.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class GoogleSignInRequest {
    @NotBlank(message = "ID token is required")
    private String idToken;

    private String serverAuthCode;
    private String email;
    private String displayName;
    private String photoUrl;
}

File: App/lib/src/features/auth/data/models/google_sign_in_request.dart
class GoogleSignInRequest {
  final String idToken;
  final String? serverAuthCode;
  final String? email;
  final String? displayName;
  final String? photoUrl;

  GoogleSignInRequest({
    required this.idToken,
    this.serverAuthCode,
    this.email,
    this.displayName,
    this.photoUrl,
  });

  Map<String, dynamic> toJson() {
    return {
      'idToken': idToken,
      if (serverAuthCode != null) 'serverAuthCode': serverAuthCode,
      if (email != null) 'email': email,
      if (displayName != null) 'displayName': displayName,
      if (photoUrl != null) 'photoUrl': photoUrl,
    };
  }
}
